PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/endpoint_app.cpp

```diff
@@ -0,0 +1,27 @@
+#include "endpoint_app.h"
+
+EndpointApp::EndpointApp(uint64_t hostMemorySize, uint64_t deviceMemorySize) {
+    memoryMap_ = std::make_unique<MemoryMap>(hostMemorySize, deviceMemorySize);
+    memoryController_ = std::make_unique<MemoryController>(std::move(memoryMap_));
+    configurationController_ = std::make_unique<ConfigurationController>();
+}
+
+
+void EndpointApp::receivePacket(TLP* packet) {
+    if (packet->header->TLPtype == TLPType::MemRead32 || packet->header->TLPtype == TLPType::MemRead64
+        || packet->header->TLPtype == TLPType::MemWrite32 || packet->header->TLPtype == TLPType::MemWrite64) {
+        memoryController_->handleTlp(packet);
+    }
+    else if (packet->header->TLPtype == TLPType::ConfigRead0 || packet->header->TLPtype == TLPType::ConfigRead1
+            || packet->header->TLPtype == TLPType::ConfigWrite0 || packet->header->TLPtype == TLPType::ConfigWrite1) {
+        configurationController_->handleConfigurationRequest(packet);
+    }
+    else {
+        // If TLP type is not recognized, do nothing and return
+        return;
+    }
+}
+
+void EndpointApp::sendPacket(TLP* packet) {
+    // Implementation of sending packets
```

Comment:
Here you didn't make any implementation, is that another story or something

Reply:
yes here this will be just passing the packet (completion) back to the layers when finishing our logic, so yes it will be another story

Reply:
You didn't accept that (functions with no implementation) when I did it, but okay I'll pass it.